src/stream/src/executor/backfill/arrangement_backfill.rs

```diff
@@ -695,8 +695,10 @@ where
             let backfill_progress = backfill_state.get_progress(&vnode)?;
             let current_pos = match backfill_progress {
                 BackfillProgressPerVnode::NotStarted => None,
-                BackfillProgressPerVnode::Completed { current_pos, .. }
-                | BackfillProgressPerVnode::InProgress { current_pos, .. } => {
```

Comment:
How did it affect the correctness of scaling background ddl? It looks like reading a `Completed` vnode only results in an empty iterator but no correctness

Reply:
This exact sequence of events (tested in https://github.com/risingwavelabs/risingwave/pull/20613):
1. 3 cns started, run backfill job.
2. 2 out of 3 cns finished their backfill.
3. pause rate limit.
4. Ingest a large amount of data, to increase the snapshot again.
5. Kill the cluster.
6. Start the cluster again, but with 1CN to trigger scale in.
7. Now we read more records into completed vnode partitions, because of the line above.
8. Subsequently, when we call `update_pos_by_vnode` for snapshot_read we panic, because it should only be called for vnodes with unfinished progress, but it's called for completed vnodes as well, since they yield more records.